sources/packages/backend/libs/services/src/notifications/notification/notification-actions.service.ts

```diff
@@ -129,12 +145,29 @@ export class GCNotifyActionsService {
     const notificationsIds = await this.notificationService.saveNotifications(
       notificationsToSend,
       auditUserId,
-      entityManager,
+      options?.entityManager,
     );
 
-    // TODO: Temporary code to be removed once queue/schedulers are in place.
-    for (const id of notificationsIds) {
-      await this.notificationService.sendEmailNotification(id);
+    if (!options?.notificationsDelayed) {
+      // Execute all the notifications in parallel and return the promisees
+      // to allow the method to await them all.
+      const notificationPromisees = notificationsIds.map((notificationId) =>
+        // Not intended to be used for large volume of notifications. If a large
+        // volume is expected, the notificationsDelayed must be set to true
+        this.notificationService.sendEmailNotification(
+          notificationId,
+          options?.entityManager,
+        ),
+      );
+
+      try {
+        // Wait all promisees.
```

Comment:
"wait for all promises."

Reply:
Changed.

Reply:
in this case we need the "for"... in "wait for something" 
https://dictionary.cambridge.org/grammar/british-grammar/wait-or-wait-for

Reply:
The reason why I added the comment on https://github.com/bcgov/SIMS/pull/1565#discussion_r1029856948 was to provide a general understanding of the reason why we do it to allow us to ponderate how far we should go.
From the moment that we start to talk more about the grammar than the code itself is when I guess that we start to lose it a bit 😉

Reply:
For me it's fine. Don't worry. I would not comment about it if you were not changing that line.